File: javascript/src/models/applicationIn.ts
// this file is @generated

export interface ApplicationIn {
  /** Application name for human consumption. */
  name: string;
  /**
   * Maximum messages per second to send to this application.
   *
   * Outgoing messages will be throttled to this rate.
   */
  throttleRate?: number | null;
  /** Optional unique identifier for the application. */
  uid?: string | null;
  metadata?: { [key: string]: string };
}

export const ApplicationInSerializer = {
  _fromJsonObject(object: any): ApplicationIn {
    return {
      name: object["name"],
      throttleRate: object["throttleRate"],
      uid: object["uid"],
      metadata: object["metadata"],
    };
  },

  _toJsonObject(self: ApplicationIn): any {
    return {
      name: self.name,
      throttleRate: self.throttleRate,
      uid: self.uid,
      metadata: self.metadata,
    };
  },
};

File: rust/src/models/application_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ApplicationIn {
    /// Application name for human consumption.
    pub name: String,

    /// Maximum messages per second to send to this application.
    ///
    /// Outgoing messages will be throttled to this rate.
    #[serde(rename = "throttleRate")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub throttle_rate: Option<u16>,

    /// Optional unique identifier for the application.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub uid: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<std::collections::HashMap<String, String>>,
}

impl ApplicationIn {
    pub fn new(name: String) -> Self {
        Self {
            name,
            throttle_rate: None,
            uid: None,
            metadata: None,
        }
    }
}
